src/graphql/root/subscription/realtime-price.ts

```diff
@@ -0,0 +1,165 @@
+      throw new UnknownClientError({
+        message:
+          "Got 'undefined' payload. Check url used to ensure right websocket endpoint was used for subscription.",
+        level: "fatal",
+        logger: baseLogger,
+      })
+    }
+
+    const { errors, timestamp, centsPerSat, centsPerUsdCent, displayCurrency } = source
+    if (errors) return { errors: errors }
+    if (!timestamp || !centsPerSat || !centsPerUsdCent || !displayCurrency) {
+      return { errors: [{ message: "No price info" }] }
+    }
+
+    const { currency } = args.input
+    if (currency instanceof Error) throw currency
+
+    if (displayCurrency !== currency) {
+      throw new UnknownClientError({
+        message: "Got 'invalid' payload.",
+        level: "fatal",
+        logger: baseLogger,
+      })
+    }
+
+    return {
+      errors: [],
+      realtimePrice: {
+        timestamp: new Date(timestamp),
+        denominatorCurrency: currency,
+        btcSatPrice: {
+          base: Math.round(centsPerSat * 10 ** SAT_PRICE_PRECISION_OFFSET),
+          offset: SAT_PRICE_PRECISION_OFFSET,
+          currencyUnit: `${currency}CENT`,
+        },
+        usdCentPrice: {
+          base: Math.round(centsPerUsdCent * 10 ** USD_PRICE_PRECISION_OFFSET),
+          offset: USD_PRICE_PRECISION_OFFSET,
+          currencyUnit: `${currency}CENT`,
+        },
+      },
+    }
+  },
+  subscribe: async (_: unknown, args: RealtimePriceSubscribeArgs) => {
+    const { currency: displayCurrency } = args.input
+
+    const immediateTrigger = customPubSubTrigger({
+      event: PubSubDefaultTriggers.PriceUpdate,
+      suffix: crypto.randomUUID(),
+    })
+
+    if (displayCurrency instanceof Error) {
+      pubsub.publishImmediate({
+        trigger: immediateTrigger,
+        payload: { errors: [{ message: displayCurrency.message }] },
+      })
+      return pubsub.createAsyncIterator({ trigger: immediateTrigger })
+    }
+
+    const currencies = await Prices.listCurrencies()
+    if (currencies instanceof Error) {
+      pubsub.publishImmediate({
+        trigger: immediateTrigger,
+        payload: { errors: [mapError(currencies)] },
+      })
+      return pubsub.createAsyncIterator({ trigger: immediateTrigger })
+    }
+
+    const priceCurrency = currencies.find((c) => displayCurrency === c.code)
+    const checkedDisplayCurrency = checkedToDisplayCurrency(priceCurrency?.code)
+
+    if (checkedDisplayCurrency instanceof Error) {
+      pubsub.publishImmediate({
+        trigger: immediateTrigger,
+        payload: { errors: [{ message: "Unsupported exchange unit" }] },
+      })
+      return pubsub.createAsyncIterator({ trigger: immediateTrigger })
+    }
+
+    const pricePerSat = await Prices.getCurrentSatPrice({
+      currency: checkedDisplayCurrency,
+    })
+    const pricePerUsdCent = await Prices.getCurrentUsdCentPrice({
+      currency: checkedDisplayCurrency,
+    })
+    if (!(pricePerSat instanceof Error) && !(pricePerUsdCent instanceof Error)) {
```

Comment:
What happens in case where `pricePerSat` or `pricePerUsdCent` is an Error. Is subscription silent?

Reply:
this is just the first time so the worst case is to wait 30 seconds for trigger update